trunner/target/armv7a9.py

```diff
@@ -19,26 +19,56 @@
 )
 from trunner.harness import TerminalHarness
 from trunner.host import Host
-from trunner.tools import Phoenixd, OpenocdGdbServer, wait_for_vid_pid
+from trunner.tools import Phoenixd, OpenocdGdbServer, Openocd, wait_for_vid_pid
 from trunner.types import TestResult, TestOptions
 from .base import TargetBase, find_port
 
 
 class ARMv7A9TargetRebooter(Rebooter):
     # TODO add text mode reboot
 
+    def __call__(self, flash=False, hard=False):
+        """Sets flash mode and perform hard or soft & debugger reboot based on `hard` flag."""
+
+        self._set_flash_mode(flash)
+
+        if hard and self.host.has_gpio():
+            self._reboot_dut_gpio(hard=hard)
+        else:
+            self._reboot_by_debugger()
```

Comment:
In such a situation with hard reboot, could we provide text instructions for the user on how to proceed with the campaign manually. If they don’t want to set up the Raspberry Pi?

Reply:
That's why we check self.host.has_gpio to utilize in this case reboot by debugger.

Reply:
As far as I know, it currently works correctly on the nucleo. However, on the zedboard, a manual hard reboot is required.  I would suggest either adding a TODO  or something like:
```suggestion
        if hard:
        	if self.host.has_gpio():
            		self._reboot_dut_gpio(hard=hard)
            	else:
            		print("Do a hard reboot.")
            		answer = input("If you perform reboot write Y")[0]
            		if answer != "Y":
            			raise Excpetion
        else:
            self._reboot_by_debugger()
```

Reply:
I am not 100% sure whether this hard reboot with power off is necessary after changing boot mode. Running the hw targets using a laptop host will be one step closer after this change, but it's still not supported on armv7a9 target, because changing boot modes is missing. I'd suggest sticking to the current solution.

Reply:
Understand. I would still consider an exception indicating that this is not supported